src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -44,12 +44,19 @@ Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
           task_spec,
           [this, actor_id, task_id](Status status, const rpc::CreateActorReply &reply) {
             if (status.ok()) {
-              RAY_LOG(DEBUG) << "Created actor, actor id = " << actor_id;
-              // Copy the actor's reply to the GCS for ref counting purposes.
-              rpc::PushTaskReply push_task_reply;
-              push_task_reply.mutable_borrowed_refs()->CopyFrom(reply.borrowed_refs());
-              task_finisher_->CompletePendingTask(
-                  task_id, push_task_reply, reply.actor_address());
+              if (reply.cancelled()) {
+                RAY_LOG(DEBUG) << "Actor creation cancelled, actor id = " << actor_id;
+                task_finisher_->MarkTaskCanceled(task_id);
+                RAY_UNUSED(task_finisher_->FailOrRetryPendingTask(
+                    task_id, rpc::ErrorType::TASK_CANCELLED, nullptr));
```

Comment:
Can you use `ACTOR_CREATION_FAILED `? 

`TASK_CANCELLED ` raises
```
class TaskCancelledError(RayError):
    """Raised when this task is cancelled.

    Attributes:
        task_id (TaskID): The TaskID of the function that was directly
            cancelled.
    """

    def __init__(self, task_id=None):
        self.task_id = task_id

    def __str__(self):
        if self.task_id is None:
            return "This task or its dependency was cancelled by"
        return "Task: " + str(self.task_id) + " was cancelled"
```
and actor creation == task is an implementation detail. (maybe we can raise an exception saying the actor scheduling is cancelled). You can also add an useful error message by providing `const rpc::RayErrorInfo *ray_error_info,` to `FailOrRetryPendingTask `.

Reply:
Ah good point!